### ISSUE TITLE
`ActionsOrganizationSecret` causes `panic: interface conversion: interface {} is string, not int` when `selectedRepositoryIds` are updated

### ISSUE THREAD

**Issue by gsuess** [pulumi/pulumi-github#249]
### What happened?

Adding to `selectedRepositoryIds` for `ActionsOrganizationSecret` causes panic: `panic: interface conversion: interface {} is string, not int`

### Steps to reproduce

Run `pulumi up` with:

```ts
import * as github from "@pulumi/github";

const repo = new github.Repository("test-repo", {
    description: "Delete me",
});

const anotherRepo = new github.Repository("another-test-repo", {
    description: "Delete me",
});

const secret = new github.ActionsOrganizationSecret(
    "testSecret",
    {
        secretName: "testSecret",
        plaintextValue: "my secret",
        visibility: "selected",
        selectedRepositoryIds: [repo.repoId],
    },
    {
        deleteBeforeReplace: true,
    }
);

```

Then add another repository ID and run `pulumi preview`

```ts
const secret = new github.ActionsOrganizationSecret(
    "testSecret",
    {
        secretName: "testSecret",
        plaintextValue: "my secret",
        visibility: "selected",
        selectedRepositoryIds: [repo.repoId, anotherRepo.repoId],
    },
    {
        deleteBeforeReplace: true,
    }
);
```

### Expected Behavior

Shows Preview without errors

### Actual Behavior
```
   Type                                       Name                   Plan       Info
     pulumi:pulumi:Stack                        pulumi-playground-dev             26 messages
 ~   ├─ github:index:Repository                 test-repo              update     [diff: -vulnerabilityAlerts]
 +   ├─ github:index:Repository                 another-test-repo      create     
     └─ github:index:ActionsOrganizationSecret  testSecret                        1 error
 
Diagnostics:
  pulumi:pulumi:Stack (pulumi-playground-dev):
    panic: interface conversion: interface {} is string, not int
    goroutine 131 [running]:
    github.com/hashicorp/terraform-plugin-sdk/helper/schema.HashInt({0x103bc6500?, 0x103e5dac0?})
        /home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.9.1/helper/schema/set.go:23 +0x5c
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v1.v1Schema.SetHash({0x140008517c0?}, {0x103bc6500, 0x103e5dac0})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.28.0/pkg/tfshim/sdk-v1/schema.go:176 +0x48
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.visitPropertyValue({0x1400084eb10, 0x17}, {0x1400084e828, 0x15}, {{0x103ba2c20?, 0x140008d6f48?}}, {0x103e79878?, 0x1400029fe00}, 0x0, 0x0, ...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.28.0/pkg/tfbridge/diff.go:114 +0x5b0
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.doIgnoreChanges({0x103e737d8, 0x140007b0720}, 0x103e737d8?, 0x103e6f720?, 0x14000b32b80?, {0x0, 0x0, 0x14000b30184?}, {0x103e751c8, 0x14000851040})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.28.0/pkg/tfbridge/diff.go:260 +0x250
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Diff(0x140002f26c0, {0x103e6e568?, 0x1400083faa0?}, 0x1400083a460)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.28.0/pkg/tfbridge/provider.go:762 +0x474
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Diff_Handler.func1({0x103e6e568, 0x1400083faa0}, {0x103da02a0?, 0x1400083a460})
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.38.0/proto/go/provider.pb.go:3707 +0x74
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x103e6e568, 0x1400083f980}, {0x103da02a0, 0x1400083a460}, 0x14000850d40, 0x140008d6e70)
        /home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x30c
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Diff_Handler({0x103e1a580?, 0x140002f26c0}, {0x103e6e568, 0x1400083f980}, 0x1400083a3f0, 0x140007a6be0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.38.0/proto/go/provider.pb.go:3709 +0x138
    google.golang.org/grpc.(*Server).processUnaryRPC(0x1400003c1e0, {0x103e755e0, 0x140005016c0}, 0x1400084d0e0, 0x140009a1320, 0x104874668, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1295 +0x9c4
    google.golang.org/grpc.(*Server).handleStream(0x1400003c1e0, {0x103e755e0, 0x140005016c0}, 0x1400084d0e0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:1636 +0x82c
    google.golang.org/grpc.(*Server).serveStreams.func1.2()
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:932 +0x84
    created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.48.0/server.go:930 +0x290
 
  github:index:ActionsOrganizationSecret (testSecret):
    error: error reading from server: EOF
 

warning: A new version of Pulumi is available. To upgrade from version '3.39.2' to '3.40.2', run 
   $ brew upgrade pulumi
or visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

### Output of `pulumi about`
```
CLI          
Version      3.39.2
Go Version   go1.19
Go Compiler  gc

Plugins
NAME    VERSION
github  4.17.0
nodejs  unknown

Host     
OS       darwin
Version  12.6
Arch     arm64

This project is written in nodejs: executable='/Users/gsuess/.volta/bin/node' version='v18.7.0'

Current Stack: dev

TYPE                                                              URN
pulumi:pulumi:Stack                                               urn:pulumi:dev::pulumi-playground::pulumi:pulumi:Stack::pulumi-playground-dev
pulumi:providers:github                                           urn:pulumi:dev::pulumi-playground::pulumi:providers:github::default_4_17_0
github:index/repository:Repository                                urn:pulumi:dev::pulumi-playground::github:index/repository:Repository::test-repo
github:index/actionsOrganizationSecret:ActionsOrganizationSecret  urn:pulumi:dev::pulumi-playground::github:index/actionsOrganizationSecret:ActionsOrganizationSecret::testSecret


Found no pending operations associated with dev

Backend        
Name           pulumi.com
URL            https://app.pulumi.com/gsuess
User           gsuess
Organizations  gsuess, Qualifyze

Dependencies:
NAME            VERSION
@types/node     14.18.31
@pulumi/github  4.17.0
@pulumi/pulumi  3.40.3

Pulumi locates its logs in /var/folders/_j/n3zrsq6n373c_18tsgvfs2yc0000gn/T/ by default
warning: A new version of Pulumi is available. To upgrade from version '3.39.2' to '3.40.2', run 
   $ brew upgrade pulumi
or visit https://pulumi.com/docs/reference/install/ for manual instructions and release notes.
```

### Additional context

This is most likely another issue at the terraform provider, but I could not find an issue reported there. Reporting it here first for reference.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by viveklak** [comment]
This is a duplicate of #157  and has also been reported previously in #111.
With v5.0.0 this no longer panics but the second update causes the following error:
```
 github:index:ActionsOrganizationSecret (testSecret):
    error: Resource provider reported that the resource did not exist while updating urn:pulumi:dev2::github::github:index/actionsOrganizationSecret:ActionsOrganizationSecret::testSecret.

    This is usually a result of the resource having been deleted outside of Pulumi, and can often be fixed by running `pulumi refresh` before updating.
```

As @guineveresaenger identified in https://github.com/pulumi/pulumi-github/issues/111#issuecomment-1108857174 - the bug is in the terraform provider.

**Comment by viveklak** [comment]
To add more color, the issue is that [`updated_at` is only updated at `Read`](https://cs.github.com/integrations/terraform-provider-github/blob/05a8875f0a3912d7231b06cfe5dc5d55bbc1dccc/github/resource_github_actions_organization_secret.go#L202). It works in the case of the initial creation since no `updated_at` is available in state (and the embedded call to `read` places the current time in `updated_at`. Subsequent update triggers the so-called drift-detection mechanism constantly because the `updated_at` field for the secret in Github doesn't match what was recorded in state. 

A quick workaround is to add the [`replaceOnChanges: ["*"]`](https://www.pulumi.com/docs/intro/concepts/resources/options/replaceonchanges/) field as a resource option. This will recreate the resource on any changes.

I will reopen this to track embedding this directly in the pulumi schema. However, since the current provider doesn't result in a panic, dropping p1/panic labels.

**Comment by gsuess** [comment]
Thanks @viveklak. I have attempted the `replaceOnChanges` workaround before, but it still panicked. I guess I will have to also upgrade to v5.0.0.

**Comment by vubogovich** [comment]
This should be fixed in [the terraform provider](https://github.com/integrations/terraform-provider-github/issues/1208) which is released in [v5.4.0](https://github.com/integrations/terraform-provider-github/releases/tag/v5.4.0). The latest pulumi provider v5.0.0 uses terraform one v5.2.0. Looking forward to the new version.
